KGD input file fix: add Fracture into the targetRegions in the SolidM…

taojinllnl opened the pull request:
…echanicsLagrangianSSLE block

Without the modification, we would have the following error at the beginning of the simulation:
Running simulation
Time: 0s, dt:0.2s, Cycle: 0
***** ERROR
***** LOCATION: /usr/WS1/jin10/geosx_dev/GEOSX/src/coreComponents/linearAlgebra/DofManager.cpp:1676
***** Controlling expression (should be false): std::find( rowRegions.begin(), rowRegions.end(), regionName ) == rowRegions.end()
***** Rank 2: Region 'Fracture' does not belong to the domain of field 'TotalDisplacement'

cssherman commented:
It looks like there is a typo in the solid mechanics solver

This is the original:

```xml
<SolidMechanicsLagrangianSSLE
      name="lagsolve"
      timeIntegrationOption="QuasiStatic"
      logLevel="0"
      discretization="FE1"
      targetRegions="{ Region2 }"
      solidMaterialNames="{ rock }"
      contactRelationName="fractureContact">

The correct version should mention the fracture region (I assume for the contact mechanics?)

<SolidMechanicsLagrangianSSLE
      name="lagsolve"
      timeIntegrationOption="QuasiStatic"
      logLevel="0"
      discretization="FE1"
      targetRegions="{ Region2, Fracture }"
      solidMaterialNames="{ rock }"
      contactRelationName="fractureContact">
```

taojinllnl commented:
> It looks like there is a typo in the solid mechanics solver
> 
> This is the original:
> 
> ```
> <SolidMechanicsLagrangianSSLE
>       name="lagsolve"
>       timeIntegrationOption="QuasiStatic"
>       logLevel="0"
>       discretization="FE1"
>       targetRegions="{ Region2 }"
>       solidMaterialNames="{ rock }"
>       contactRelationName="fractureContact">
> 
> The correct version should mention the fracture region (I assume for the contact mechanics?)
> 
> <SolidMechanicsLagrangianSSLE
>       name="lagsolve"
>       timeIntegrationOption="QuasiStatic"
>       logLevel="0"
>       discretization="FE1"
>       targetRegions="{ Region2, Fracture }"
>       solidMaterialNames="{ rock }"
>       contactRelationName="fractureContact">
> ```

Yep, this is the issue. And I modified the xml accordingly.

francoishamon commented:
@taojinllnl @cssherman Sorry for breaking your input file with PR #1156, I forgot to update the files in the `examples` folder, but I updated the integrated tests files. 

To apply `SolidMechanicsLagrangianFEM` on target regions more consistently and solve #1140 , I added the target regions names to the `addField` call to `dofManager`:
https://github.com/GEOSX/GEOSX/blob/e44d96546da77188ff788bdc8ec42834f22b9216/src/coreComponents/physicsSolvers/solidMechanics/SolidMechanicsLagrangianFEM.cpp#L961-L964
To make this work for some coupled solvers (namely, `LagrangianContactSolver` and `HydrofractureSolver`), this change required specifying `targetRegions="{ Region, Fracture }"` instead of just `targetRegions="{ Region }"` in the `SolidMechanicsLagrangianFEM` block of the xml file. See the discussion in issue #1140 about this.

taojinllnl commented:
> @taojinllnl @cssherman Sorry for breaking your input file with PR #1156, I forgot to update the files in the `examples` folder, but I updated the integrated tests files.
> 
> To apply `SolidMechanicsLagrangianFEM` on target regions more consistently and solve #1140 , I added the target regions names to the `addField` call to `dofManager`:
> https://github.com/GEOSX/GEOSX/blob/e44d96546da77188ff788bdc8ec42834f22b9216/src/coreComponents/physicsSolvers/solidMechanics/SolidMechanicsLagrangianFEM.cpp#L961-L964
> 
> 
> To make this work for some coupled solvers (namely, `LagrangianContactSolver` and `HydrofractureSolver`), this change required specifying `targetRegions="{ Region, Fracture }"` instead of just `targetRegions="{ Region }"` in the `SolidMechanicsLagrangianFEM` block of the xml file. See the discussion in issue #1140 about this.

No worries at all. I did some reverse engineering and figured out the issue. I have updated the KGD xml file in the example folder  in this PR. @rrsettgast
